Alkousa/gcn_tele.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import gcn
+import logging
+import telegram
+import datetime as dt
+
+try:
+    with open('./token') as f:
+        my_token = f.read().strip()
+except OSError:
+    print('no token, can\'t work')
+    exit(1)
+try:
+    with open('./proxy') as f:
+        url, un, pw = f.read().strip().split(',')
+        REQUEST_KWARGS = {
+            'proxy_url': '{}'.format(url),
+            # Optional, if you need authentication:
+            'urllib3_proxy_kwargs': {
+                'username': '{}'.format(un),
+                'password': '{}'.format(pw),
+            }
+        }
+        req = telegram.utils.request.Request(**REQUEST_KWARGS)
+except OSError:
+    req = telegram.utils.request.Request()
+    print('no proxy file, not using it')
+
+
+def send(msg, chat_id, token=my_token):
+    bot = telegram.Bot(token=token, request=req)
+    bot.sendMessage(chat_id=chat_id, text=msg, parse_mode='Markdown')
+
+
+def gcn_handler(payload, root):
+    # if root.attrib['role'] == 'test':
+    # return
+    print(root.attrib['ivorn'])
+    msg = root.attrib['ivorn']
+    params = {elem.attrib['name']:
+              elem.attrib['value']
+              for elem in root.iterfind('.//Param')}
+    for key, value in params.items():
+        print(key, '=', value)
+        msg = '{msg}\n\
+        **{key}** = {value}'.format(msg=msg, key=key, value=value)
+    try:
+        pos2d = root.find('.//{*}Position2D')
+        ra = float(pos2d.find('.//{*}C1').text)
+        dec = float(pos2d.find('.//{*}C2').text)
+        radius = float(pos2d.find('.//{*}Error2Radius').text)
+        print('ra = {:g}, dec={:g}, radius={:g}'.format(ra, dec, radius))
+        msg = '{msg}\n\
+        Coord data available!\n\
+        RA={ra}\n\
+        DE={de}\n\
+        EB={eb}\n'.format(msg=msg, ra=ra, de=dec, eb=radius)
+    except AttributeError:
+        print("no coords")
+        msg = msg + '\nNo coord data'
+    send(msg, '@g_c_n_alert', my_token)
+
+
+if __name__ == "__main__":
+    logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level=logging.INFO)
+    send('GCN Alert system starting at {now}'.format(now=dt.datetime.now().isoformat()), '@g_c_n_alert', my_token)
```

Comment:
I don't talk about API key, a talk about this line where `@g_c_n_alert` is hardcoded. It should be at least some configurable thing. Practically Telegram bots can work with any number of chats and channels without server-side reconfiguration, see `@GitHubBot` as example

Reply:
Yes, in that regard, when it comes to a channel name, it should also be hard-coded.

Reply:
But than how do another bots like `@GitHubBot` work?

Reply:
I understand the thing about channels, but what about group chats?

Reply:
It could sounds like a good idea to add a bot like this to SAI group chat or some other chats

Reply:
I couldn't figure out how these kinds of bots work. One way to go around this is what I tried to explain in the email, but that is still considered a hard-code, because I am not obtaining channel @Handles automatically, but rather creating some sort of a list of channels, which is not different from hard coding the names.

Reply:
Ok, can you just put this channel name to some separated file or environment variable?

Reply:
sure!

Reply:
changed the code and modified README.md